packages/core/src/network/heartbeat.ts

```diff
@@ -142,10 +140,7 @@ export default class Heartbeat {
     let pingResponse: Uint8Array[] | undefined
 
     try {
-      pingResponse = await this.sendMessage(destination, this.protocolHeartbeat, challenge, true, {
-        timeout: this.config.heartbeatDialTimeout,
-        signal
-      })
+      pingResponse = await this.sendMessage(destination, this.protocolHeartbeat, challenge, true)
```

Comment:
Ping can now take forever?

Reply:
There is a global timeout mechanism which is the same for any connect attempt and for every of our protocol selection and for every of our message interaction, so tldr it will time out